packages/backend/jobs-manager/service/src/modules/database/subscriptions/cron-subscriptions/cron-subscriptions.service.ts

```diff
@@ -306,8 +318,18 @@ export class CronSubscriptionsService {
     finding: Finding,
     projectId: string,
   ) {
-    if (!SubscriptionsUtils.shouldExecuteFromFinding(sub.conditions, finding))
+    if (
+      !SubscriptionsUtils.shouldExecuteFromFinding(
+        sub.isEnabled,
+        sub.conditions,
+        finding,
+      )
+    ) {
+      this.logger.debug(
+        `Skipping job publication for ${sub.name}; conditions not met or subscription is disabled.`,
+      );
       return;
```

Comment:
At this point, we did a whole lot of operations just to be denied at the end. Kinda feels like they should be denied at the beginning of the workflow to save a whole lot of operations.  (see previous comment)